runtime/tests/Test_LightningGateSet.cpp

```diff
@@ -478,6 +478,63 @@ TEMPLATE_LIST_TEST_CASE("CSWAP test", "[GateSet]", SimTypes)
     CHECK(state[5].imag() == Approx(0).epsilon(1e-5));
 }
 
+/*
+TEMPLATE_LIST_TEST_CASE("ISWAP test", "[GateSet]", SimTypes)
+{
+    std::unique_ptr<TestType> sim = std::make_unique<TestType>();
+
+    // state-vector with #qubits = n
+    constexpr size_t n = 2;
+    std::vector<QubitIdType> Qs;
+    Qs.reserve(n);
+
+    for (size_t i = 0; i < n; i++) {
+        Qs[i] = sim->AllocateQubit();
+    }
+
+    sim->NamedOperation("PauliX", {}, {Qs[0]}, false);
+    sim->NamedOperation("ISWAP", {}, {Qs[0], Qs[1]}, false);
+
+    std::vector<std::complex<double>> state(1U << sim->GetNumQubits());
+    DataView<std::complex<double>, 1> view(state);
+    sim->State(view);
+
+    CHECK(state.at(0) == std::complex<double>{0, 0});
+    CHECK(state.at(1) == std::complex<double>{0, 1});
+    CHECK(state.at(2) == std::complex<double>{0, 0});
+    CHECK(state.at(3) == std::complex<double>{0, 0});
+}
+
+TEMPLATE_LIST_TEST_CASE("PSWAP test", "[GateSet]", SimTypes)
+{
+    std::unique_ptr<TestType> sim = std::make_unique<TestType>();
+
+    // state-vector with #qubits = n
+    constexpr size_t n = 2;
+    std::vector<QubitIdType> Qs;
+    Qs.reserve(n);
+
+    for (size_t i = 0; i < n; i++) {
+        Qs[i] = sim->AllocateQubit();
+    }
+
+    sim->NamedOperation("Hadamard", {}, {Qs[1]}, false);
+    sim->NamedOperation("PSWAP", {M_PI_2}, {Qs[0], Qs[1]}, false);
+    sim->NamedOperation("Hadamard", {}, {Qs[0]}, false);
+
+    std::vector<std::complex<double>> state(1U << sim->GetNumQubits());
+    DataView<std::complex<double>, 1> view(state);
+    sim->State(view);
+
+    CHECK(state[0].real() == Approx(0.5).epsilon(1e-5));
+    CHECK(state[0].imag() == Approx(0.5).epsilon(1e-5));
+    CHECK(state.at(1) == std::complex<double>{0, 0});
+    CHECK(state[2].real() == Approx(0.5).epsilon(1e-5));
+    CHECK(state[2].imag() == Approx(-0.5).epsilon(1e-5));
+    CHECK(state.at(3) == std::complex<double>{0, 0});
+}
+*/
```

Comment:
You can keep these tests and use `REQUIRE_THROWS_WITH` to check they are failed with some key-value exceptions. I'd be okay if you leave them as is too. We can revisit them after adding the patch upstream to the pennylane-lightning repository,

Reply:
I will keep them as is. I believe in some other test files (e.g. `test_operations.py`) there are commented tests waiting to be uncommented once the support arrives.